src/handlers/secret.rs

```diff
@@ -67,18 +70,33 @@ pub async fn create_secret_handler(
     attempt_rule: &State<AttemptCountRule>,
     secret: Json<NewSecret>,
 ) -> Result<Json<Secret>, status::Custom<Json<Value>>> {
-    let created_secret = storage
-        .create_secret(secret.into_inner())
-        .await;
+    let secret_data = NewSecret {
+        secret: secret.secret.to_lowercase(),
+        clue1: secret.clue1.to_lowercase(),
+        clue2: secret.clue2.to_lowercase(),
+        clue3: secret.clue3.to_lowercase(),
```

Comment:
I don't think the clues also need to be lowercased, since we won't be comparing anything with them.